src/main/java/org/jitsi/videobridge/cc/SimulcastController.java

```diff
@@ -787,23 +787,23 @@ public boolean accept(
                     this.currentIdx = currentIdx;
                 }
 
-                boolean isNewest = maxSentFrame == null
-                    || TimeUtils.rtpDiff(srcTs, maxSentFrame.srcTs) > 0;
+                boolean isNewerThanMostRecentFrame = !haveSentFrame()
+                    || RTPUtils.isNewerTimestampThan(srcTs, mostRecentSentFrame.srcTs);
 
-                boolean isNewerThanMostRecentKeyFrame = mrKeyFrame == null
-                    || TimeUtils.rtpDiff(srcTs, mrKeyFrame.srcTs) > 0;
+                boolean isNewerThanMostRecentKeyFrame = !haveSentKeyFrame()
+                    || RTPUtils.isNewerTimestampThan(srcTs, mostRecentSentKeyFrame.srcTs);
 
-                if (currentIdx > -1
+                if (!isSuspended()
                     // we haven't seen anything yet and this is an independent
                     // frame.
-                    && (maxSentFrame == null && sourceFrameDesc.isIndependent()
+                    && (!haveSentFrame() && sourceFrameDesc.isIndependent()
```

Comment:
In the comment for haveSentFrame you wrote "has sent a frame" but the method is called "haveSentFrame." I think if both George and I felt a little off by the naming then it's worth re-examining to make sure it makes as much sense as you want.